source/slang/slang-ir-autodiff.cpp

```diff
@@ -399,6 +399,17 @@ void DifferentiableTypeConformanceContext::setFunc(IRGlobalValueWithCode* func)
             else
             {
                 differentiableWitnessDictionary.Add((IRType*)item->getConcreteType(), item->getWitness());
+
+                // Also register the type's differential type with the same witness.
+                IRBuilder subBuilder(item->getConcreteType());
+                // TODO: What do we do for second order derivatives through interfaces?
```

Comment:
I believe _lookupWitness will emit a `LookupWitness` inst if the witnss operand isn't a known witness table.